src/shared/components/person/profile.tsx

```diff
@@ -300,224 +332,173 @@ export class Profile extends Component<ProfileRouteProps, ProfileState> {
   }
 
   componentWillReceiveProps(nextProps: ProfileRouteProps) {
-    // Overview, Posts and Comments views can use the same data.
-    const sharedViewTypes = [nextProps.view, this.props.view].every(
-      v =>
-        v === PersonDetailsView.Overview ||
-        v === PersonDetailsView.Posts ||
-        v === PersonDetailsView.Comments,
-    );
-
     const reload = bareRoutePush(this.props, nextProps);
 
     const newUsername =
       nextProps.match.params.username !== this.props.match.params.username;
 
     if (
-      (nextProps.view !== this.props.view && !sharedViewTypes) ||
+      nextProps.view !== this.props.view ||
       nextProps.sort !== this.props.sort ||
-      nextProps.page !== this.props.page ||
+      nextProps.cursor !== this.props.cursor ||
+      nextProps.filter !== this.props.filter ||
       newUsername ||
       reload
     ) {
       this.fetchUserData(nextProps, reload || newUsername);
     }
   }
 
-  fetchUploadsToken?: symbol;
-  async fetchUploads(props: ProfileRouteProps) {
-    if (!this.amCurrentUser) {
-      return;
-    }
-
-    const token = (this.fetchUploadsToken = Symbol());
-    const { page } = props;
-    this.setState({ uploadsRes: LOADING_REQUEST });
-    const form: ListMedia = {
-      // userId?
-      page,
-      limit: fetchLimit,
-    };
-    const uploadsRes = await HttpService.client.listMedia(form);
-    if (token === this.fetchUploadsToken) {
-      this.setState({ uploadsRes });
-    }
-  }
-
   fetchUserDataToken?: symbol;
   async fetchUserData(props: ProfileRouteProps, showBothLoading = false) {
-    const token = (this.fetchUploadsToken = this.fetchUserDataToken = Symbol());
-    const { page, sort, view } = props;
-
-    const requestLiked = view === PersonDetailsView.Upvoted;
+    const token = (this.fetchUserDataToken = Symbol());
```

Comment:
This could probably be neater by creating separate functions, like `fetchUserLikes`, `fetchUserSaved`, rather than trying to do them all at once.

We should basically only show/fetch the content by default, and let every other tab do live fetches.

Can be done later tho.